### ISSUE TITLE
False positive with keywords-only fields of dataclasses when using defaulted fields

### ISSUE THREAD

**Issue by NeilGirdhar** [PyCQA/pylint#7881]
### Bug description

This is a more complicated example than https://github.com/PyCQA/pylint/issues/5767
```python
from dataclasses import KW_ONLY, dataclass, field

@dataclass
class Plotter:
    node_names: list[str] = field(default_factory=list)

@dataclass
class RewardsPlotter(Plotter):
    _: KW_ONLY
    smoothing: float

RewardsPlotter(smoothing=4.0)  # Unexpected keyword argument
```

### Pylint output

```shell
a.py:13:0: E1123: Unexpected keyword argument 'smoothing' in constructor call (unexpected-keyword-arg)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.12.13
Python 3.10.4 (main, Apr  7 2022, 10:39:08) [GCC 11.2.0]
```
